FullBackgroundAudio/app/src/main/java/com/jwplayer/demo/fullbackgroundaudio/BackgroundAudioService.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Author: Efrain Gonzalez
+ * c-egonzalez@jwplayer.com
+ */
+
+package com.jwplayer.demo.fullbackgroundaudio;
+
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+import android.support.v4.media.MediaMetadataCompat;
+import android.support.v4.media.session.MediaSessionCompat;
+import android.support.v4.media.session.PlaybackStateCompat;
+import android.widget.Toast;
+
+import com.longtailvideo.jwplayer.JWPlayerView;
+import com.longtailvideo.jwplayer.configuration.PlayerConfig;
+import com.longtailvideo.jwplayer.core.PlayerState;
+import com.longtailvideo.jwplayer.events.PauseEvent;
+import com.longtailvideo.jwplayer.events.PlayEvent;
+import com.longtailvideo.jwplayer.events.listeners.VideoPlayerEvents;
+import com.longtailvideo.jwplayer.media.playlists.PlaylistItem;
+
+
+public class BackgroundAudioService extends Service {
+
+    private static final float PLAYBACK_SPEED = 1.0f;
+
+    private JWPlayerView mPlayer;
+
+    private MediaSessionCompat mMediaSessionCompat;
+
+    private PlayerEventHandler eventHandler = new PlayerEventHandler();
+
+    private PlaybackStateCompat.Builder mPlaybackStateBuilder = new PlaybackStateCompat.Builder();
+    private MediaMetadataCompat.Builder mMediaMetadataBuilder = new MediaMetadataCompat.Builder();
+
+    private ServiceBinder mBinder = new ServiceBinder();
+
+
+    @Override
+    public void onCreate() {
+        mMediaSessionCompat = new MediaSessionCompat(this, getClass().getSimpleName());
+        mMediaSessionCompat.setCallback(new MediaSessionCallback());
+        mMediaSessionCompat.setActive(true);
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        String action = intent.getStringExtra("ACTION");
+        switch (action) {
+            case "ACTION_START": {
+                //START FOREGROUND SERVICE
+                NotificationCompat.Builder builder = new NotificationCompat.Builder(this, App.CHANNEL_ID);
+                builder.setContentTitle(mPlayer.getPlaylistItem().getTitle())
```

Comment:
Coding style here: I think it's better to have one method call per line, and the subsequent lines should be indented.

Reply:
You could also pull the `PlaylistItem` into a local variable since you're using it a few times here.

Reply:
Pulled the Playlist item into a local variable